connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -793,6 +796,38 @@ protected void assignConnectors(List<WorkerLoad> workerAssignment, Collection<St
         }
     }
 
+    static class BalancedIterator<E> implements Iterator<E> {
+
+        Map<String, Iterator<E>> grouped;
+        List<String> keys;
+        int k;
```

Comment:
Nit: can reduce visibility and make some fields final
```suggestion
        private final Map<String, Iterator<E>> grouped;
        private final List<String> keys;

        private int k;
```